File: platform/src/main/java/e_commerce/platform/admin/service/AdminCategoryService.java
package e_commerce.platform.admin.service;

import e_commerce.platform.modules.category.dto.request.CreateCategoryRequest;
import e_commerce.platform.modules.category.dto.request.UpdateCategoryRequest;
import e_commerce.platform.modules.category.dto.response.CategoryResponse;
import org.springframework.data.domain.Page;

public interface AdminCategoryService {

    Page<CategoryResponse> getAll(int page, int size);

    CategoryResponse create(CreateCategoryRequest request);

    CategoryResponse update(Long id, UpdateCategoryRequest request);

    void delete(Long id);
}

File: ecommerce-frontend/src/features/admin/services/adminCategoryService.ts
import axiosInstance from "@/config/axios";
import type {
  ApiResponse,
  PageResponse,
  AdminCategory,
  CreateCategoryRequest,
  UpdateCategoryRequest,
} from "../types/adminTypes";

export const adminCategoryService = {
  async getAll(page = 0, size = 20) {
    const res = await axiosInstance.get<ApiResponse<PageResponse<AdminCategory>>>(
      `/api/admin/categories?page=${page}&size=${size}`
    );
    return res.data.data;
  },

  async create(data: CreateCategoryRequest) {
    const res = await axiosInstance.post<ApiResponse<AdminCategory>>(
      `/api/admin/categories`,
      data
    );
    return res.data.data;
  },

  async update(id: number, data: UpdateCategoryRequest) {
    const res = await axiosInstance.put<ApiResponse<AdminCategory>>(
      `/api/admin/categories/${id}`,
      data
    );
    return res.data.data;
  },

  async delete(id: number) {
    await axiosInstance.delete(`/api/admin/categories/${id}`);
  },
};